ens-improvement-proposals/ens-l2-offchain.md

```diff
@@ -0,0 +1,117 @@
+
+- [EIP-3668: CCIP Read: Secure offchain data retrieval](https://eips.ethereum.org/EIPS/eip-3668)
+- [ENSIP 10: Wildcard Resolution](ens-improvement-proposals/ensip-10-wildcard-resolution)
+- [ENSIP-11: EVM compatible Chain Address Resolution](ens-improvement-proposals/ensip-11-evmchain-address-resolution.md)
+
+**EIP 3668** allows for offchain (including Layer 2/L2) lookups of data in a way that is transparent to clients and provides contract authors to implement whatever validation is necessary; in many cases, this can be provided without any additional trust assumptions over and above those required if data is stored onchain.
+
+**ENSIP 10** is a general way to resolve wildcard (eg: *.foo.eth) on L1. Issuing subdomains and moving the resolution of the parent name offchain allows dapps to create subdomains offchain yet make it accessible through L1.
+
+**ENSIP 11** allows a single name to hold different addresses across multiple EVM compatible chains. Contract addresses can use these EVM chain addresses so that clients can choose appropriate addresses on different chains.
+
+## Steps required for Dapps and wallets to support offchain data lookup.
+
+If your dapps or wallets use one of those libraries, the EIP 3668 and ENSIP 10 support will be built in, so simply update the library when ready.
+
+- [ethersjs](https://github.com/ethers-io/ethers.js)
+- [web3js](https://github.com/ethereum/web3.js)
+
+### ethersjs
+
+Currently EIP 3668 is implemented as npm module as [@chainlink/ethers-ccip-read-provider
+](@chainlink/ethers-ccip-read-provider
+)([source](https://github.com/smartcontractkit/ccip-read/tree/rewrite/packages/ethers-ccip-read-provider)).
+
+The basic usage example is as follows.
+
+```js
+import { CCIPReadProvider } from '@chainlink/ethers-ccip-read-provider';
+import { abi as IExtendedResolver_abi } from '@ensdomains/offchain-resolver-contracts/artifacts/contracts/IExtendedResolver.sol/IExtendedResolver.json';
+const IExtendedResolver = new ethers.utils.Interface(IExtendedResolver_abi);
+const baseProvider = ethers.getDefaultProvider(options.provider);
+const provider = new CCIPReadProvider(baseProvider);
+```
+
+The wildcard is still not supported yet, but one can easily be implemented as follows.
```

Comment:
I think this snippet is not very useful on its own - especially with 'Resolver' and 'registry' undefined. I don't think we should be encouraging people to resolve names like this.

Reply:
what is the recommendation? If it's not ready I can just say the support is work in progress

Reply:
I think that's the best option right now.